.circleci/config.yml

```diff
@@ -209,11 +246,50 @@ jobs:
           command: ghr -t $GITHUB_TOKEN -u $CIRCLE_PROJECT_USERNAME -r $CIRCLE_PROJECT_REPONAME --replace $CIRCLE_TAG bin/
 
   publish-dev:
-    docker:
-      - image: cimg/go:1.14
+    executor: golang
     steps:
       - attach_to_workspace
       - setup_remote_docker
       - publish_docker_images:
           repo: opentelemetry-collector-contrib-dev
           tag: ${CIRCLE_SHA1}
+
+  spawn-stability-tests-job:
+    executor: golang
+    steps:
+      - run:
+          name: Trigger stability tests job
+          command: |
+            curl -f -X POST "https://circleci.com/api/v2/project/github/open-telemetry/${CIRCLE_PROJECT_REPONAME}/pipeline?circle-token=${CIRCLE_API_TOKEN}" \
+                -H 'Content-Type: application/json' \
+                -H 'Accept: application/json' \
+                -d '{"parameters": {"run-build-publish": false, "run-stability-tests": true, "collector-sha": "'"${CIRCLE_SHA1}"'"}, "branch": "'"${CIRCLE_BRANCH}"'"}'
+
+  checkout-commit:
+    executor: golang
+    steps:
+      - checkout
+      - run:
+          name: Checkout pipeline parameter commit
+          command: |
+            git checkout << pipeline.parameters.collector-sha >>
+            git status
+
+  run-stability-tests:
+    executor: golang
+    steps:
+      - run:
+          name: Run stability tests
+          command: make stability-tests
+      - run:
+          name: Run on fail status
+          command: |
+              curl --request POST \
+              --url https://api.github.com/repos/open-telemetry/opentelemetry-collector-contrib/issues \
+              --header "authorization: Bearer ${GITHUB_TOKEN}" \
+              --header "content-type: application/json" \
+              --data '{
+                "title": "Stability tests failed in branch '"${CIRCLE_BRANCH}"' for commit << pipeline.parameters.collector-sha >>",
+                "body": "Link to failed job: '"${CIRCLE_BUILD_URL}"'."
```

Comment:
Maybe have a label for these to be able to filter by?

Reply:
I don't think the token I created can add labels to issues. If we want to add that, I think the token will need push access which means CircleCI will need push access to the repo. We can still do that after we discuss usage of tokens and permissions with the rest of the group. For starters, we need to create some bot github accounts that can own the tokens and decide whether CI having push access to the repo is acceptable or not.